source/client/service_impl.cc

```diff
@@ -0,0 +1,104 @@
+#include "client/service_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include "client/client.h"
+#include "client/options_impl.h"
+
+#include "api/client/options.pb.validate.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void ServiceImpl::nighthawkRunner(const nighthawk::client::ExecutionRequest& request) {
+  Envoy::Thread::LockGuard lock(mutex_);
+  OptionsPtr options = std::make_unique<OptionsImpl>(request.options());
+  Envoy::Thread::MutexBasicLockable log_lock;
+  auto logging_context = std::make_unique<Envoy::Logger::Context>(
+      spdlog::level::from_str(options->verbosity()), "[%T.%f][%t][%L] %v", log_lock);
+  process_ = std::make_unique<ProcessImpl>(*options, time_system_);
+
+  // We perform this validation here because we need to rutime to be initialized
+  // for this, something that ProcessContext does for us.
+  try {
+    Envoy::MessageUtil::validate(request.options());
+  } catch (Envoy::EnvoyException exception) {
+    nighthawk::client::ExecutionResponse response;
+    response_queue_.push(ServiceProcessResult(response, exception.what()));
```

Comment:
Why is it necessary to queue the response? Is it the case that gRPC stubs doesn't support cross-thread replies?